### PR TITLE
feat(cloudflare): Add Snippet resource

### PR DESCRIPTION
**Pull request by JacobMGEvans**
## Summary by CodeRabbit

* **New Features**
  * Added Snippet resource for full lifecycle management of Cloudflare snippets (create, update, delete, list) with name validation, content sourcing, and adoption.
  * Added SnippetRule resource for expression-based snippet rules with adopt, enable/disable, ordering, and lifecycle operations.
  * Expanded public surface to expose snippet and snippet-rule functionality.

* **Tests**
  * Added comprehensive end-to-end tests covering Snippet and SnippetRule lifecycles, adoption, edge cases, and performance scenarios.

* **Documentation**
  * Added usage examples and guidance for snippet and rule workflows.

### CODE REVIEW

File: alchemy/src/cloudflare/snippet.ts [hunk 1]
```diff
@@ -0,0 +1,430 @@
...
+   * Description of what this rule does
+   */
+  description?: string;
+
+  /**
+   * Whether this rule is enabled
+   * @default true
+   */
+  enabled?: boolean;
+}
```
Comment (john-royal) [inline alchemy/src/cloudflare/snippet.ts]: Question: I see `SnippetRule` referenced in some functions here and tests. I don't see a resource for it though. How is this meant to be used?

File: alchemy/src/cloudflare/snippet.ts [hunk 2]
```diff
@@ -0,0 +1,430 @@
...
+            `/zones/${zoneId}/snippets/${this.output.name}`,
+          );
+          if (!deleteResponse.ok && deleteResponse.status !== 404) {
+            await handleApiError(
+              deleteResponse,
+              "delete",
+              "snippet",
+              this.output.name,
+            );
+          }
```
Comment (sam-goodwin) [inline alchemy/src/cloudflare/snippet.ts]: Each of these should be extracted as re-usable API functions and placed below the resource

File: alchemy/src/cloudflare/snippet.ts (hunk 1 above)
Comment (JacobMGEvans) [reply alchemy/src/cloudflare/snippet.ts]: https://developers.cloudflare.com/api/node/resources/snippets/subresources/rules/

File: alchemy/src/cloudflare/snippet.ts [hunk 3]
```diff
@@ -0,0 +1,466 @@
...
+   * The identifying name of the snippet
+   * Must be unique within the zone
+   * Only lowercase letters (a-z), numbers (0-9), and underscores (_) are allowed
+   */
+  name: string;
+
+  /**
+   * The JavaScript code content of the snippet
+   */
+  content: string;
```
Comment (sam-goodwin) [inline alchemy/src/cloudflare/snippet.ts]: Should we support `entrypoint` like we do for Worker?

File: alchemy/src/cloudflare/snippet.ts [hunk 4]
```diff
@@ -0,0 +1,466 @@
...
+   * The zone to create the snippet in
+   */
+  zone: string | Zone;
+
+  /**
+   * The identifying name of the snippet
+   * Must be unique within the zone
+   * Only lowercase letters (a-z), numbers (0-9), and underscores (_) are allowed
+   */
+  name: string;
```
Comment (sam-goodwin) [inline alchemy/src/cloudflare/snippet.ts]: Should this be optional and by default derived from the `id`? 

```ts
const name = props.name ?? this.scope.createPhysicalName(id)
```

File: alchemy/src/cloudflare/snippet.ts [hunk 5]
```diff
@@ -0,0 +1,466 @@
...
+   * The JavaScript code content of the snippet
+   */
+  content: string;
+
+  /**
+   * Whether to delete the snippet
+   * If set to false, the snippet will remain but the resource will be removed from state
+   * @default true
+   */
+  delete?: boolean;
```
Comment (sam-goodwin) [inline alchemy/src/cloudflare/snippet.ts]: Curious why this use-case exists for Snippet but not Worker?

File: alchemy/src/cloudflare/snippet.ts [hunk 6]
```diff
@@ -0,0 +1,466 @@
...
+   * The Cloudflare-generated zone ID (only used for update/delete operations)
+   * @internal
+   */
+  zoneId?: string;
+
+  /**
+   * The snippet name (only used for update/delete operations)
+   * @internal
+   */
+  snippetId?: string;
```
Comment (sam-goodwin) [inline alchemy/src/cloudflare/snippet.ts]: This should not be in the props, right? We have it in state and it should be encapsulated.

File: alchemy/src/cloudflare/snippet.ts [hunk 7]
```diff
@@ -0,0 +1,466 @@
...
+   * Whether to adopt an existing snippet
+   * @default false
+   */
+  adopt?: boolean;
+
+  /**
+   * The Cloudflare-generated zone ID (only used for update/delete operations)
+   * @internal
+   */
+  zoneId?: string;
```
Comment (sam-goodwin) [inline alchemy/src/cloudflare/snippet.ts]: `zone: string | Zone`

File: alchemy/src/cloudflare/snippet.ts (hunk 6 above)
Comment (sam-goodwin) [reply alchemy/src/cloudflare/snippet.ts]: This should be on the output, not the input props

File: alchemy/src/cloudflare/snippet.ts (hunk 4 above)
Comment (sam-goodwin) [reply alchemy/src/cloudflare/snippet.ts]: Are dashes allowed?

File: alchemy/src/cloudflare/snippet.ts [hunk 8]
```diff
@@ -0,0 +1,466 @@
...
+    `list snippet rules in zone "${zoneId}"`,
+    api.get(`/zones/${zoneId}/snippets/snippet_rules`),
+  );
+}
+
+/**
+ * Update snippet rules in a zone
+ * @internal
+ */
+export async function updateSnippetRules(
```
Comment (sam-goodwin) [inline alchemy/src/cloudflare/snippet.ts]: This function is only used in tests?

File: alchemy/src/cloudflare/snippet.ts [hunk 9]
```diff
@@ -0,0 +1,466 @@
...
+ *     }
+ *   `
+ * });
+ *
+ * @example
+ * // Create a snippet that modifies request URLs
+ * const urlRewriteSnippet = await Snippet("url-rewrite", {
+ *   zone: "example.com",
+ *   name: "rewrite-api-urls",
+ *   content: `
```
Comment (sam-goodwin) [inline alchemy/src/cloudflare/snippet.ts]: in Worker, it's `script | entrypoint`. I kinda regret those names but maybe keep it consistent

**Review by sam-goodwin — COMMENTED** [review]
Looks like `SnippetRule` should be supported as input props?

Also missing docs

File: alchemy/src/cloudflare/snippet.ts (hunk 4 above)
Comment (JacobMGEvans) [reply alchemy/src/cloudflare/snippet.ts]: No, underscores, letters & numbers only

**Comment by Mkassabov** [comment]
@coderabbitai can you review this please?

**Comment by Mkassabov** [comment]
@coderabbitai Hey we changed some of the review rules. can you review this again?

**Comment by Mkassabov** [comment]
> @Mkassabov: I'll review the changes again with the updated review rules.
> ✅ Actions performed
> 
> Review triggered.
> 
> > Note: CodeRabbit is an incremental review system and does not re-review already reviewed commits. This command is applicable only when automatic reviews are paused.

Hey @coderabbitai its changed now. rereview please!

**Comment by JacobMGEvans** [comment]
@coderabbitai review
